compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -5305,14 +5305,14 @@ abstract class GenJSCode extends plugins.PluginComponent
 
   /** Tests whether the given class is a Scala.js-defined JS class. */
   def isScalaJSDefinedJSClass(sym: Symbol): Boolean =
-    !sym.isTrait && sym.hasAnnotation(ScalaJSDefinedAnnotation)
+    !sym.isTrait && isRawJSType(sym.tpe) && !sym.hasAnnotation(JSNativeAnnotation)
 
   def isScalaJSDefinedAnonJSClass(sym: Symbol): Boolean =
     sym.hasAnnotation(SJSDefinedAnonymousClassAnnotation)
 
   /** Tests whether the given class is a JS native class. */
   private def isJSNativeClass(sym: Symbol): Boolean =
-    isRawJSType(sym.tpe) && !isScalaJSDefinedJSClass(sym)
+    isRawJSType(sym.tpe) && sym.hasAnnotation(JSNativeAnnotation)
```

Comment:
I think that `@js.native` implies `@RawJSType`, so this could be simplified as only `sym.hasAnnotation(JSNativeAnnotation)`.